packages/store/src/usersSlice/thunks.ts

```diff
@@ -746,30 +867,40 @@ export const getTokens = createAsyncThunk(`${sliceName}/getTokens`, (_, { dispat
 
 const ONE_YEAR = 31536000;
 
-export const generateToken = createAsyncThunk(`${sliceName}/generateToken`, ({ expiresIn = ONE_YEAR, name }, { dispatch }) =>
-  GeneralApi.post(`${useradmApiUrl}/settings/tokens`, { name, expires_in: expiresIn })
-    .then(({ data: token }) => Promise.all([dispatch(getTokens()), token]))
-    .catch(err => commonErrorHandler(err, 'There was an error creating the token:', dispatch))
+export const generateToken = createAppAsyncThunk(
+  `${sliceName}/generateToken`,
+  ({ expiresIn = ONE_YEAR, name }: { expiresIn?: number; name: string }, { dispatch }) =>
```

Comment:
There is `import type { PersonalAccessTokenRequest } from '@northern.tech/store/api/types/MenderTypes';` already.
```suggestion
  ({ expiresIn = ONE_YEAR, name }: PersonalAccessTokenRequest, { dispatch }) =>
```

Reply:
It's `expires_in` in `PersonalAccessTokenRequest`, but this function accepts object with `expiresIn` property. Such a change would require more changes in Mender gui for debatable gains